classes/Village.php

```diff
@@ -129,4 +135,287 @@ public function getRelations(System $system, string $village_id): array {
+                "user_name" => null,
+                "avatar_link" => null,
+            ];
+        }
+        return $seats;
+    }
+
+    /**
+     * @return array
+     */
+    public static function getResources(System $system, int $village_id): array
+    {
+        $resources_result = $system->db->query("SELECT `resources` FROM `villages` WHERE `village_id` = {$village_id}");
+        $resources_result = $system->db->fetch($resources_result);
+        return json_decode($resources_result['resources'], true);
+    }
+    /**
+     * @return array
+     */
+    public static function getResourceHistory(System $system, int $village_id, int $days): array
+    {
+        $resource_history = [];
+        $time = time() - $days * 86400;
+        foreach (array_keys(WarManager::RESOURCE_NAMES) as $resource_id) {
+            // get produced
+            $result = $system->db->query("SELECT SUM(`quantity`) as 'produced' FROM `resource_logs` WHERE `resource_id` = {$resource_id} AND `village_id` = {$village_id} AND `type` = " . Village::RESOURCE_LOG_PRODUCTION . " AND `time` > {$time}");
+            $result = $system->db->fetch($result);
+            $resource_history[$resource_id]['produced'] = (int)$result['produced'];
+            // get collected
+            $result = $system->db->query("SELECT SUM(`quantity`) as 'collected' FROM `resource_logs` WHERE `resource_id` = {$resource_id} AND `village_id` = {$village_id} AND `type` = " . Village::RESOURCE_LOG_COLLECTION . " AND `time` > {$time}");
+            $result = $system->db->fetch($result);
+            $resource_history[$resource_id]['collected'] = (int)$result['collected'];
+            // get claimed
+            $result = $system->db->query("SELECT COUNT(*) as 'claimed' FROM `loot` WHERE `resource_id` = {$resource_id} AND `claimed_village_id` = {$village_id} AND `claimed_time` > {$time}");
+            $result = $system->db->fetch($result);
+            $resource_history[$resource_id]['claimed'] = (int)$result['claimed'];
+            // get lost (lol)
+            // can determine resources lost based on difference of produced and collected
+            $resource_history[$resource_id]['lost'] = $resource_history[$resource_id]['produced'] - $resource_history[$resource_id]['collected'];
+            // factor in resources which haven't been collected from regions
+            $result = $system->db->query("SELECT SUM(`resource_count`) as 'quantity' FROM `region_locations` INNER JOIN `regions` on `regions`.`region_id` = `region_locations`.`region_id` WHERE `resource_id` = {$resource_id} AND `village` = {$village_id}");
+            $result = $system->db->fetch($result);
+            $resource_history[$resource_id]['lost'] -= (int) $result['quantity'];
+            // get upkeep - WIP
+            $resource_history[$resource_id]['spent'] = 0;
+        }
+        // extra step to calculate lost resources: factor in resources which haven't been collected from caravans
```

Comment:
```suggestion
        // extra step to calculate lost resources: don't include resources which haven't been collected from caravans as lost
```
I was initially a little confused, does this wording match your intent? Reading the code below it looks like this is what's going on

Reply:
Summary of what's going on
- We know how many resources have been produced
- We know how many resources have been collected
- [Produced - Collected] = Lost, in other words the difference is how many resources other villages have stolen

There are two exceptions to this
- Some resources are still sitting in villages waiting to be collected
- Some resources are currently held in caravans (the comment for the section here)

I'm not 100% sure on how to word this best as a comment, what do you recommend?

Reply:
That makes sense - I think for this part the rewording in my suggestion matches what you described